jujugui/static/gui/src/app/app.js

```diff
@@ -411,7 +411,35 @@ YUI.add('juju-gui', function(Y) {
         interactive: this.get('interactiveLogin'),
         serviceName: 'juju',
         dischargeStore: window.localStorage,
-        dischargeToken: dischargeToken
+        dischargeToken: dischargeToken,
+        visitMethod: response => {
```

Comment:
No reason, just used the method from the original prototype from Huw. I hope that we are able to remove all this code asap, when implementing the proper login with redirection flow.